### PR TITLE
Compile under CentOS 7.8

### PR DESCRIPTION
**Pull request by tuxmaster5000**
Fix for
https://github.com/greenbone/gvm-libs/issues/395

### CODE REVIEW

File: boreas/CMakeLists.txt [hunk 1]
```diff
@@ -58,6 +58,13 @@ set (FILES alivedetection.c boreas_error.c boreas_io.c cli.c ping.c sniffer.c ut
 
 set (HEADERS alivedetection.h boreas_error.h boreas_io.h cli.h ping.h sniffer.h util.h)
 
+if (CMAKE_C_COMPILER_ID STREQUAL "GNU")
+  if(${GLIB_VERSION} VERSION_LESS "2.50.3")
+    add_definitions(-D_BSD_SOURCE)
+  endif()
+  add_compile_options(-std=gnu99)
```
Comment (mattmundell) [inline boreas/CMakeLists.txt:65]:
```suggestion
    add_compile_options(-std=gnu99)
```

File: boreas/CMakeLists.txt [hunk 2]
```diff
@@ -58,6 +58,13 @@ set (FILES alivedetection.c boreas_error.c boreas_io.c cli.c ping.c sniffer.c ut
 
 set (HEADERS alivedetection.h boreas_error.h boreas_io.h cli.h ping.h sniffer.h util.h)
 
+if (CMAKE_C_COMPILER_ID STREQUAL "GNU")
+  if(${GLIB_VERSION} VERSION_LESS "2.50.3")
+    add_definitions(-D_BSD_SOURCE)
+  endif()
```
Comment (mattmundell) [inline boreas/CMakeLists.txt:64]:
```suggestion
  endif(${GLIB_VERSION} VERSION_LESS "2.50.3")
```

File: boreas/CMakeLists.txt [hunk 3]
```diff
@@ -58,6 +58,13 @@ set (FILES alivedetection.c boreas_error.c boreas_io.c cli.c ping.c sniffer.c ut
...
+  add_compile_options(-std=gnu99)
+endif()
```
Comment (mattmundell) [inline boreas/CMakeLists.txt:66]:
```suggestion
endif(CMAKE_C_COMPILER_ID STREQUAL "GNU")
```

**Review by mattmundell — CHANGES_REQUESTED** [review]
Thanks.

Do you want this in 20.08.x?  If so we should change the destination branch to greenbone:gvm-libs-20.08.

And please add a CHANGELOG.md entry.

**Comment by mattmundell** [comment]
Also, would be great to have a little comment about why this is required.

**Comment by tuxmaster5000** [comment]
> Also, would be great to have a little comment about why this is required.

See the the issue link, why it is needed. It looks like the code was only tested again an very new version of glibc.

**Comment by bjoernricks** [comment]
> See the the issue link, why it is needed. It looks like the code was only tested again an very new version of glibc.

Our base system is Debian as you can see from our CI images. Therefore nearly all Distributions based on Debian will work without any issues. We can't invest our resources for supporting all kind of other distributions. Therefore we rely on external contributors like you to provide patches if something breaks.

**Comment by tuxmaster5000** [comment]
> > See the the issue link, why it is needed. It looks like the code was only tested again an very new version of glibc.
> 
> Our base system is Debian as you can see from our CI images. Therefore nearly all Distributions based on Debian will work without any issues. We can't invest our resources for supporting all kind of other distributions. Therefore we rely on external contributors like you to provide patches if something breaks.

This is no problem,  because it is not my first patch for the greenbone world. :)

**Review by bjoernricks — APPROVED** [review]

**Pull request merged**